client/state/imports/actions.js

```diff
@@ -98,32 +98,33 @@ export function cancelImport( siteId, importerId ) {
 		return;
 	}
 
-	apiStart();
-	wpcom
+	dispatch( apiStart() );
+
+	return wpcom
 		.updateImporter( siteId, cancelOrder( siteId, importerId ) )
-		.then( apiSuccess )
+		.then( apiSuccess( dispatch ) )
 		.then( fromApi )
-		.then( receiveImporterStatus )
-		.catch( apiFailure );
-}
+		.then( receiveImporterStatus( dispatch ) )
+		.catch( apiFailure( dispatch ) );
```

Comment:
With classic action creators, this code can become:
```js
wpcom.updateImporter( ... ).then( data => {
  dispatch( apiSuccess() );
  dispatch( receiveImporterStatus( fromApi( data ) ) );
} ).catch( error => {
  dispatch( apiFailure() );
} );
```

As a next step, can we merge the `IMPORTS_FETCH_COMPLETED` and `IMPORTS_IMPORT_RECEIVE` actions into one? They are always dispatched together.

Reply:
…and we can skip all this by using `trackRequests` request-tracking